File: group1/frontend/src/request/LoginRequest.ts
/**
 * Represents a request to log into a user account
 */
export class LoginRequest {
    public isGuest: boolean;
    public username?: string;

    constructor(isGuest: boolean, username?: string) {
        this.isGuest = isGuest;
        this.username = username;
    }
}

File: group1/backend/src/main/java/com/group1/backend/request/LoginRequest.java
package com.group1.backend.request;

import org.springframework.lang.Nullable;

/**
 * Represents a login request from the frontend
 */
public class LoginRequest {
    public boolean isGuest;
    public @Nullable String username;

    /**
     * @return true if the user is a guest, false if they are logged in
     */
    public boolean isGuest() {
        return isGuest;
    }

    /**
     * @return the username of the user, or null if they are a guest
     */
    @Nullable
    public String getUsername() {
        return username;
    }

    @Override
    public String toString() {
        return "LoginRequest{" +
                "isGuest=" + isGuest +
                ", username='" + username + '\'' +
                '}';
    }
}
